test/assembler/linker.test.ts

```diff
@@ -169,7 +178,71 @@ describe('linker', function () {
     expect(Halfword.fromBytes(file.content[12], file.content[13])).toEqual(
       END_OF_CODE
     )
-    expect(file.content[14]).toEqual(Byte.fromUnsignedInteger(0x22))
+    expect(Halfword.fromBytes(file.content[14], file.content[15])).toEqual(
+      Halfword.fromUnsignedInteger(0x0000)
+    )
+    expect(file.content[16]).toEqual(Byte.fromUnsignedInteger(0x22))
+    expect(file.content[17]).toEqual(Byte.fromUnsignedInteger(0x00))
+    expect(file.content[18]).toEqual(Byte.fromUnsignedInteger(0x00))
+    expect(file.content[19]).toEqual(Byte.fromUnsignedInteger(0x00))
+  })
+  it('should align sections in segments', function () {
+    const code: ICodeFile = {
+      symbols: {},
+      areas: [
+        {
+          type: AreaType.Data,
+          isReadOnly: false,
+          name: '|.data_1|',
+          instructions: [
+            {
+              name: 'DCB',
+              options: ['0x22'],
+              line: 0
+            }
+          ]
+        },
+        {
+          type: AreaType.Data,
+          isReadOnly: false,
+          name: '|.data_2|',
+          instructions: [
+            {
+              name: 'DCW',
+              options: ['0x2222'],
+              line: 1
+            }
+          ]
+        },
+        {
+          type: AreaType.Code,
+          isReadOnly: true,
+          name: '|.text|',
+          instructions: [
+            {
+              name: 'MOVS',
+              options: ['R1', 'R1'],
+              line: 2
+            },
+            {
+              name: 'MOVS',
+              options: ['R2', 'R2'],
+              line: 3
+            }
+          ]
+        }
+      ]
+    }
+    const file = link(encode(code))
+    expect(file.segments.length).toBe(2)
+    expect(file.segments[0].type).toBe(SegmentType.Load)
+    expect(file.segments[0].offset).toBe(0)
+    expect(file.segments[0].size).toBe(16)
+    expect(file.segments[0].address).toEqual(Word.fromSignedInteger(0x08000000))
+    expect(file.segments[1].type).toBe(SegmentType.Load)
```

Comment:
check if data sections are written correctly aligned. maybe use other value on DCW and DCB to ensure it is not the other

Reply:
I've added additional asserts for the content to this test.